pcomfortcloud/authentication.py

```diff
@@ -242,7 +268,7 @@ def _get_new_token(self):
                 "x-app-timestamp": timestamp,
                 "x-app-type": "1",
                 "x-app-version": self._app_version,
-                "x-cfc-api-key": generate_random_string_hex(128),
+                "x-cfc-api-key": self._get_api_key(timestamp, self._token["access_token"]),
```

Comment:
This should be fixed as well
```suggestion
                "x-cfc-api-key": self._get_api_key(timestamp, token_response["access_token"]),
```

Reply:
Right, the new token should be used, where possible. However, I'm quite sure it should be the same access token as used in the Authorization header, so I've only changed one of the above.

Reply:
You are right. Great work